hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducer.java

```diff
@@ -574,13 +583,41 @@ public InputStream decompress(InputStream is) {
          * It is guaranteed that {@code blob} was created by calling one of
          * {@link BlobStager#openSnapshot(long)}, {@link BlobStager#openDelta(long, long)}, or
          * {@link BlobStager#openReverseDelta(long, long)} on this publisher.
+         * {@link BlobStager#openHeader(long)} on this publisher.
          *
-         * @param blob the blob to publish
+         * @param publishArtifact the blob to publish
          */
-        void publish(HollowProducer.Blob blob);
+        void publish(HollowProducer.AbstractPublishArtifact publishArtifact);
+    }
+
+    public static abstract class AbstractPublishArtifact {
```

Comment:
An interface would be better suited here

Reply:
Done